lib/core/src/recover/handlers/handle_chain_send_swap.rs

```diff
@@ -0,0 +1,258 @@
+use anyhow::Result;
+use log::{debug, error, warn};
+use lwk_wollet::elements::Txid;
+
+use crate::prelude::*;
+use crate::recover::model::*;
+
+/// Handler for updating chain send swaps with recovered data
+pub(crate) struct ChainSendSwapHandler;
+
+impl ChainSendSwapHandler {
+    /// Check if chain send swap recovery should be skipped
+    pub fn should_skip_recovery(
+        chain_swap: &ChainSwap,
+        recovered_data: &RecoveredOnchainDataChainSend,
+        is_local_within_grace_period: bool,
+    ) -> bool {
+        let swap_id = &chain_swap.id;
+
+        let lockup_is_cleared = chain_swap.user_lockup_tx_id.is_some()
+            && recovered_data.lbtc_user_lockup_tx_id.is_none();
+        let refund_is_cleared =
+            chain_swap.refund_tx_id.is_some() && recovered_data.lbtc_refund_tx_id.is_none();
+
+        if is_local_within_grace_period && (lockup_is_cleared || refund_is_cleared) {
```

Comment:
I see you removed the check for `claim_is_cleared`. Could you elaborate on why we don't need to check it?

Reply:
Dropped during the refactor, I added it back.